### DISCUSSION TITLE
Application theming

### DISCUSSION THREAD

**Discussion by FrancescoCaracciolo** [end-4/dots-hyprland#540] [General]
I wanted to theme other applications (like Telegram, Vesktop and Obsidian) using the generated color scheme. 
It would be pretty easy to theme with Pywal, however the Pywal color scheme is not automatically generated on wallpaper change.
So I had 3 questions related to this problem:
1) Is there any way to "adapt" the generated color scheme to a pywal color scheme to make other applications follow it?
2) Is the pywal color scheme actually generated somewhere? (Since it is a dependency)
3) Is there an easy way to execute a script after wallpaper change? (So that I can actually generate color schemes and apply them on my own, even if they slightly differ)

**Comment by end-4** [comment]
1. In `~/.local/state/ags/scss/_material.scss` scroll down to the end and you'll see colors $term0, $term1, $term2, etc. Use that however you wish
2. Pywal is used only for the music widget since readability isn't critical on that widget
3. Easy or not, you tell me.... Just add whatever you wanna run to the script (`~/.config/ags/scripts/color_generation/applycolor.sh`) or something

**Reply by FrancescoCaracciolo** [reply]
Thank you, that was exactly what I was searching for.
I think you should add a script in user space called every time the wallpaper is changed that doesn't get overridden with updates.

**Reply by RawnakHasan** [reply]
Were you able to theme obsidian? It would be great if you could help me to do that as well
